### ISSUE TITLE
bug(youtube/swipe-controls): not working on first swipe

### ISSUE THREAD

**Issue by oSumAtrIX** [revanced/revanced-patches#814]

**Comment by SodaWithoutSparkles** [comment]
No video was attached. I will attach my own.

https://user-images.githubusercontent.com/64138578/197333313-90b6666d-3af6-448c-955c-826d1a0d2eaf.mp4

I can reproduce the issue. This issue can be mitigated by repeated tapping until the video controls (play/pause buttons) dont appear before swiping.

Note that a few hundred ms had to be waited after the buttons disappear by itself, as to prevent youtube thinking you wanted to press the controls but missed it.

**Comment by Frontline11** [comment]
Yes correct there is a problem even in the first swipe swipe, this bug is old but not fixed

**Comment by Frontline11** [comment]
this bug needs to be fixed urgently

**Comment by DaniAndTheWeb** [comment]
A straight forward fix could be the removal of the video cards whenever the swipe controls are enabled in the settings

**Comment by SodaWithoutSparkles** [comment]
I consider swiping up and bringing video cards out when the okayer controls are visible is a feature and should not be removed. Just tap once and hide the controls then swipe, no big deal when you get used to it. It can be convenient sometimes.

**Comment by oSumAtrIX** [comment]
![image](https://user-images.githubusercontent.com/13122796/228612719-644870b3-3780-41aa-9523-b6d7549bceee.png)

**Comment by SodaWithoutSparkles** [comment]
It is not clear whay the issue actually is, as no videos have been posted by the orginal poster. However, if it was the one descripted in https://github.com/revanced/revanced-patches/issues/814#issuecomment-1287699856, then I don't think that this is an issue at all, at most an enhancement to an existing patch, as descripted [here](https://github.com/revanced/revanced-patches/issues/814#issuecomment-1289804443). 

Yes, the fact that you need hide the player controls before swiping to control volume is a bit confusing at first. But just because it is confusing doesn't mean the patch does not work. Just because it is a bit counter-intuitive to use, doesn't mean the design is flawed at the beginning. This keeps the og feature of being able to use the suggested video feature. Sure, some documentation might be needed, but I don't think the whole approach should be modified.

Speaking of the suggested video feature, what if you used some combination of `hide-player-buttons`, `hide-seekbar`, `hide-timestamp`, "layout > hide related videos in quick actions", "layout > hide quick actions in fullscreen" and "layout > hide previous and next buttons"? Would it mitigate this issue?

**Comment by AnonymousWP** [comment]
There's no such setting or patch as "hide previous and next buttons". Disabling 'video cards' is also not a thing. I guess they got renamed in a PR.

**Comment by AnonymousWP** [comment]
> > There's no such setting or patch as "hide previous and next buttons". Disabling 'video cards' is also not a thing. I guess they got renamed in a PR.
> 
> It is in patch `hide-player-buttons`. You need to add that patch first, then should you have the layout > hide previous and next buttons

I did, but it didn't fix the issue. I still need to swipe very precisely at the very millisecond. Also, I don't get how I just faced this bug. Before, I didn't have this issue. It's since the latest patch (169) for me I believe.

**Comment by SodaWithoutSparkles** [comment]
@AnonymousWP screen recording please. Nobody provided any screen recording from the start, I was just guessing if mine have the same issue.

**Comment by AnonymousWP** [comment]
So, I found a workaround: just tap-and-hold on the screen and _then_ swipe. Maybe I'll submit a PR when I have time to clarify this in some patch.

**Comment by AnonymousWP** [comment]
So, two concrete questions:

1. I'd like to edit the description and include a note about the above. Do I include the note in the description of "Control volume and brightness" underneath "Swipe controls"? Or do I add them separately to its respective descriptions of both separate options? E.g. "Enable brightness gesture" and "Enable volume gesture"?
2. "Swipe magnitude threshold"'s value is a bit weird. The default is 30. But 30 what? I understand that higher means more sensitive (I assume), but is it a percentage with a maximum value or just a random value?

I think, if you agree, I'll make two separate PRs.

**Comment by oSumAtrIX** [comment]
1. I am not sure if I understood correctly, what do you want to do?
2. It is a constant: 
![image](https://github.com/revanced/revanced-patches/assets/13122796/b7f07bb3-2b82-452e-8744-af60eef90c95)

**Comment by AnonymousWP** [comment]
> 1. I am not sure if I understood correctly, what do you want to do?
> 2. It is a constant:
>    ![image](https://user-images.githubusercontent.com/13122796/239392442-b7f07bb3-2b82-452e-8744-af60eef90c95.png)

Hmm, it's a bit complicated to explain my second point (I just tested this thoroughly, but I'll do my best to explain:

1. As you know, swipe controls are broken. The workaround that works for me is to touch-hold-swipe. I'd like to notify users by a note in YouTube ReVanced, so I was thinking of modifying a line in https://github.com/revanced/revanced-patches/blob/main/src/main/kotlin/app/revanced/patches/youtube/interaction/swipecontrols/patch/resource/SwipeControlsResourcePatch.kt. However, my question is: where do I put the note (which line)?
2. I sort of understand what a `constant` is, but that still makes me wonder about its context: I can set it to 0, but also to 9000. Does higher mean I need to swipe further? There's also no limit, and when I set it to 9000 for a test, it didn't work anymore at all. So I'd suggest to improve the description and a maximum value. Unlike with other descriptions (which are clear), I now don't know what the difference is between setting it to 30 or 500. I didn't notice a difference. Maybe that doesn't work either as the 'normal' way of using swipe controls (which is the core reason this GitHub issue exists) is bugged.
